klient/k8s/res/resource_test.go

```diff
@@ -80,6 +81,84 @@ func TestRes(t *testing.T) {
 	}
 }
 
+func TestResNoConfig(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Error("expected panic while invoking Res without k8s config")
+		}
+	}()
+
+	Res(nil)
```

Comment:
Random question, how come Res panics instead of returning error?

Reply:
I am also not very sure but I believe if controller runtime client is not instantiated then the normal execution of goroutine should be stopped as there is nothing that can be done without Resource object.

Reply:
Also if we consider returning error, then also the option that is left with calling method is check error is exit (as it cannot proceed with normal flow, since that requires resource)

Reply:
@tech-geek29 Yes, you are correct. If controller runtime client is not instantiated then execution should be stopped as there is nothing can be done without Resource. So we have added panic and not error handling as this will be a rare scenario.